arroyo-console/src/routes/pipelines/CreatePipeline.tsx

```diff
@@ -183,6 +185,14 @@ export function CreatePipeline() {
     setTabIndex(1);
   };
 
+  const stopPreview = async () => {
+    await updatePipeline({ stop: 'immediate' });
+
+    if (outputSource) {
```

Comment:
I wonder if instead of doing this when the user clicks "stop" we should instead be following the state machine of the pipeline so that we can also close it when the job stops on its own